dbcon/execplan/predicateoperator.cpp

```diff
@@ -216,6 +216,10 @@ void PredicateOperator::setOpType(Type& l, Type& r)
                 // should following the result type that MySQL gives
                 fOperationType = l;
                 fOperationType.scale = (l.scale > r.scale ? l.scale : r.scale);
+                fOperationType.precision = (l.precision > r.precision ?
+                    l.precision : r.precision);
```

Comment:
I'd use std::max() here:

 ` fOperationType.precision = std::max(l.precision, r.precision);`

It's shorter and more readable.